pages/a/1k1-azure-post.html

```diff
@@ -0,0 +1,239 @@
+title: Crate guest post for Azure - Setting up Azure for Crate
+author: Chris Ward
+description: Setting up a 1001 node Crate cluster on Azure with Docker Swarm
+created: 2016-03-29
+status: publish
+post_type: post
+tags: azure, arm
+category: developernews
+
+# Crate guest post for Azure - Setting up Azure for Crate
+
+[Crate.io](https://crate.io/) is striving to create the most scalable database in our industry;
+we aim to make scaling your data from millions to billions of records and beyond, as simple as
+possible.
+
+We have clients who write and query gigabytes of data a minute, store terabytes a day and love the
+performance and stability that Crate offers. But the question always remained at the back of our
+minds: when taken to the extreme, what is Crate really capable of?
+
+We specifically wanted to:
+
+- Know how Crate would distribute shards and partitions across such a large cluster.
+- Examine the overhead added by the cluster management.
+- See at what speeds the cluster could write data.
+
+Microsoft Azure generously lent us the use of 8000 cores as the infrastructure for this test and
+engineer time to guide us through problems and questions we might have.
+
+Microsoft wanted to show that their Azure platform was reliable and capable enough for a project
+that would scale rapidly and so resource intensive.
+
+
+## Plan
+
+While discussing the approach of setting up a 1001 node Crate cluster, we immediately thought about
+using [Docker Swarm](https://www.docker.com/products/docker-swarm) to launch Crate. However, we
+also wanted to see how the conventional deployment, using regular packages, would work out.
+
+### Docker Swarm setup
+
+Our initial approach was to start the Crate cluster in containerized environment using
+[Docker](https://www.docker.com), [Docker Swarm](https://www.docker.com/products/docker-swarm)
+for scheduling Crate instances and [Consul](https://www.consul.io/) for node discovery in a Swarm
+cluster. The solution has its pros and cons. Scheduling Crate nodes, and being able to change
+settings in the Crate cluster with minimal effort (in order to repeat certain experiments, by any
+means), is the strongest side of the chosen approach. On the other hand, it adds complexity to the
```

Comment:
i would celebrate this more with a sentence like: "Configuring, starting and managing Crate was as easy as you could imagine."